app/qml/map/MMMapThemeDrawer.qml

```diff
@@ -0,0 +1,227 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick
+import QtQuick.Controls
+
+import mm 1.0 as MM
+
+import "../components"
+
+Drawer {
+  id: root
+
+  visible: false
+  modal: true
+  interactive: false
+  dragMargin: 0 // prevents opening the drawer by dragging.
+
+  padding: 20
+
+  width: ApplicationWindow.window.width
+  height: ApplicationWindow.window.height / 2
+  edge: Qt.BottomEdge
+
+  Item {
+    focus: true
+    Keys.onReleased: function( event ) {
+      if (event.key === Qt.Key_Back || event.key === Qt.Key_Escape) {
+        root.close()
+      }
+    }
+  }
+
+  Rectangle {
+    color: roundedRect.color
+    anchors.top: parent.top
+    anchors.left: parent.left
+    anchors.right: parent.right
+    height: 2 * radius
+    anchors.topMargin: -radius
+    radius: __style.pageMargins
```

Comment:
Just for readable:
height: 2 * __style.pageMargins
radius: height / 2
anchors.topMargin: __style.pageMargins